### PR TITLE
Migrate `src/runtime/client-config-manager` to TypeScript

### PR DESCRIPTION
**Pull request by ChrisAntaki**

### CODE REVIEW

**Comment by ChrisAntaki** [comment]
Waiting on #2883

File: src/runtime/client-config-manager.ts [hunk 1]
```diff
@@ -176,20 +157,13 @@ export class ClientConfigManager {
         '/clientconfiguration'
     );
     const json = await this.fetcher_.fetchCredentialedJson(url);
-    if (json.errorMessages && json.errorMessages.length > 0) {
```
Comment (ChrisAntaki) [inline src/runtime/client-config-manager.ts:179]: Note: This seems unused, based on the server code.

File: src/runtime/client-config-manager.ts [hunk 2]
```diff
@@ -128,19 +112,17 @@ export class ClientConfigManager {
...
       !!this.clientOptions_.forceLangInIframes && !!this.clientOptions_.lang
     );
   }
 
   /**
    * Determines whether a subscription or contribution button should be disabled.
-   * @returns {!Promise<boolean|undefined>}
    */
-  async shouldEnableButton() {
+  async shouldEnableButton(): Promise<boolean | void> {
```
Comment (justinchou-google) [inline src/runtime/client-config-manager.ts:125]: nit: is `Promise<void>` the replacement for `Promise<undefined>`?

**Review by justinchou-google — APPROVED** [review]

File: src/runtime/client-config-manager.ts (hunk 2 above)
Comment (ChrisAntaki) [reply src/runtime/client-config-manager.ts:125]: Yep, exactly

**Pull request merged**